pkg/ottl/parser_test.go

```diff
@@ -1724,6 +1902,65 @@ func Test_parseStatement(t *testing.T) {
 	}
 }
 
+// This test doesn't validate parser results, simply checks whether the parse succeeds or not.
+// It's a fast way to check a large range of possible syntaxes.
+func Test_parseCondition(t *testing.T) {
+	tests := []struct {
+		condition string
+		wantErr   bool
+	}{
+		{`set(`, true},
+		{`set("foo)`, true},
+		{`set(name.)`, true},
+		{`("foo")`, true},
+		{`name =||= "fido"`, true},
+		{`name = "fido"`, true},
+		{`name or "fido"`, true},
+		{`name and "fido"`, true},
+		{`name and`, true},
+		{`name or`, true},
+		{`(`, true},
+		{`)`, true},
+		{`(name == "fido"))`, true},
+		{`((name == "fido")`, true},
+		{`set()`, true},
+		{`Int() == 1`, false},
+		{`1 == Int()`, false},
+		{`true and 1 == Int() `, false},
+		{`false or 1 == Int() `, false},
+		{`service == "pinger" or foo.attributes["endpoint"] == "/x/alive"`, false},
+		{`service == "pinger" or foo.attributes["verb"] == "GET" and foo.attributes["endpoint"] == "/x/alive"`, false},
+		{`animal > "cat"`, false},
+		{`animal >= "cat"`, false},
+		{`animal <= "cat"`, false},
+		{`animal < "cat"`, false},
+		{`animal =< "dog"`, true},
+		{`animal => "dog"`, true},
+		{`animal <> "dog"`, true},
+		{`animal = "dog"`, true},
+		{`animal`, true},
+		{`animal ==`, true},
+		{`==`, true},
+		{`== animal`, true},
+		{`attributes["path"] == "/healthcheck"`, false},
+		{`One() == 1`, false},
+		{`test(fail())`, true},
+		{`Test()`, false},
+	}
+	pat := regexp.MustCompile("[^a-zA-Z0-9]+")
+	for _, tt := range tests {
+		name := pat.ReplaceAllString(tt.condition, "_")
+		t.Run(name, func(t *testing.T) {
+			ast, err := parseCondition(tt.condition)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("parseCondition(%s) error = %v, wantErr %v", tt.condition, err, tt.wantErr)
+				t.Errorf("AST: %+v", ast)
+				return
+			}
+		})
+	}
+}
+
 func Test_Execute(t *testing.T) {
```

Comment:
Not a big deal, but it probably wouldn't be bad to have a case for `Condition.Eval()`.

Reply:
Good idea